File: library/src/main/java/org/incendo/interfaces/interfaces/PlayerInterfaceBuilder.java
package org.incendo.interfaces.interfaces;

import org.incendo.interfaces.pane.PlayerPane;

public final class PlayerInterfaceBuilder extends AbstractInterfaceBuilder<PlayerPane, PlayerInterface> {

    @Override
    public PlayerInterface build() {
        return new PlayerInterface(
            this.closeHandlers(),
            this.transforms(),
            this.clickPreprocessors()
        );
    }

}

File: next/src/main/kotlin/org/incendo/interfaces/next/interfaces/PlayerInterfaceBuilder.kt
package org.incendo.interfaces.next.interfaces

import org.incendo.interfaces.next.pane.PlayerPane

public class PlayerInterfaceBuilder : AbstractInterfaceBuilder<PlayerPane, PlayerInterface>() {
    override fun build(): PlayerInterface =
        PlayerInterface(
            closeHandlers,
            transforms,
            clickPreprocessors,
            itemPostProcessor,
        )
}
